### DISCUSSION TITLE
Connecting multiple systems

### DISCUSSION THREAD

**Discussion by MaartenSchild** [yamcs/yamcs#828] [Q&A]
Hi all,
For our purpose we would like to monitor multiple systems. I was looking through the documentation but I struggled to find the best practice to achieve this.

These systems make use of the same telemetry packets, how can I best distinguish them. We currently have a working instance which obtains different telemetry packets based on CCDS_Packet/APID from a single system. From what I have read this can be done with starting multiple instances, would I then create unique APIDs for each system? 

Thanks in advance for any help/pointers!

**Comment by xpromache** [comment]
What do you mean "these systems make use of the same telemetry packets"? Do they have the same APID/packet id or only the same structure?

Anyway, one way to do it is to have one instance per system but this is not very convenient because at the current time you cannot combine data on displays from multiple instances (unless you build yourself some display software).

Another way to do it is to use the same instance and have different MDB container roots together with multiple links. If you look at the simulator example there is a tm2 stream which receives data in parallel on a different link/stream. Then you define the root container used to decode the data and so it ends up populating different parameters.

**Reply by MaartenSchild** [reply]
Thank you for this. They share the same structure, so we were wondering to use the same APIDs or use different APIDS for each system regardless of the structure being the same. We don't currently have a whole lot of experience with all this.

**Reply by xpromache** [reply]
In general it is better to use a different APID because if you have a binary packet, you know where it's coming from just looking at it. As I said, in Yamcs when injecting the data it is possible to tag the packet with a different name so you can make the distinction but as soon as the packet is out of Yamcs, you don't know anymore. So better use a different APID if possible.

Now, your packets from the different systems have the same structure; in order to avoid redefining the structure for each system,  please have a look at this issue:
https://github.com/yamcs/yamcs/issues/822

Btw, SpaceApps organizes a Yamcs training: https://yamcs.org/training
It may be interesting for you. Not sure you can still book a place in this first training but I think they will organize another one if enough people are interested.

**Comment by codenamefrance** [comment]
I was kinda thinking about this problem, and I was wondering if YAMCS natively supports (or if it is easily implementable) some way of splitting different APID data in "separate visualization".
My idea is, in fact, to have n systems, each identified by an APID. I'd like to have all these systems sending data on the same stream (e.g., s1 on the instance i1). 
Then would be great to have something that allows to visualize all the data based on the APID; does YAMCS already supports this kind of behaviour?
